store/evmstore.go

```diff
@@ -315,15 +311,20 @@ func (s *EvmStore) GetSnapshot(version int64) db.Snapshot {
 }
 
 func NewEvmStoreSnapshot(snapshot db.Snapshot, rootHash []byte) *EvmStoreSnapshot {
-	return &EvmStoreSnapshot{
+	evmSnapshot := &EvmStoreSnapshot{
 		Snapshot: snapshot,
 		rootHash: rootHash,
 	}
+	ethDB := NewLoomEthDB(evmSnapshot, nil)
+	// This should be read only database
+	evmSnapshot.trieDB = trie.NewDatabase(ethDB)
```

Comment:
The evm.db snapshot will only contain the state as of the last flushed block, so this `EvmStoreSnapshot` will not have any state changes made in-memory since the last flush to disk.

Reply:
Yeah I know, it is a snapshot so the uncommitted state should be excluded

Reply:
Oh hmmmmm I see what you mean. Probably need to find a way to fix this.

Reply:
Probably need to implement this committing state once per block https://github.com/loomnetwork/loomchain/issues/1568 in this PR.